swarm/network/protocol.go

```diff
@@ -214,10 +217,25 @@ func (b *Bzz) performHandshake(p *protocols.Peer, handshake *HandshakeMsg) error
 		return err
 	}
 	handshake.peerAddr = rsh.(*HandshakeMsg).Addr
+	sanitizeEnodeRemote(p.RemoteAddr(), handshake.peerAddr)
 	handshake.LightNode = rsh.(*HandshakeMsg).LightNode
 	return nil
 }
 
+// the remote enode string may advertise arbitrary host information (e.g. localhost)
+// this method ensures that the addr of the peer will be the one
+// applicable on the interface the connection came in on
+// it modifies the passed bzzaddr in place, and returns the same pointer
+func sanitizeEnodeRemote(paddr net.Addr, baddr *BzzAddr) {
+	hsSubmatch := regexpEnodeIP.FindSubmatch(baddr.UAddr)
+	ip, _, err := net.SplitHostPort(paddr.String())
+	if err == nil && string(hsSubmatch[1]) != ip {
```

Comment:
This is a potential panic. Please add a comment that IPv4 format is expected since the check was rejected https://github.com/ethersphere/go-ethereum/pull/1224#discussion_r258349231, although this check has an insignificant cost to performance. A comment would help a lot a person that would debug such panic.

Reply:
Thanks @nolash.